packages/hardhat-core/src/internal/cli/vars.ts

```diff
@@ -0,0 +1,305 @@
+import chalk from "chalk";
+import debug from "debug";
+import { HardhatError, assertHardhatInvariant } from "../core/errors";
+import { ERRORS } from "../core/errors-list";
+import { HardhatContext } from "../context";
+import { VarsManagerSetup } from "../core/vars/vars-manager-setup";
+import {
+  importCsjOrEsModule,
+  resolveConfigPath,
+} from "../core/config/config-loading";
+import { getVarsFilePath } from "../util/global-dir";
+import { ArgumentsParser } from "./ArgumentsParser";
+import { emoji } from "./emoji";
+
+const log = debug("hardhat:cli:vars");
+
+export async function handleVars(
+  allUnparsedCLAs: string[],
+  configPath: string | undefined
+): Promise<number> {
+  const { taskDefinition, taskArguments } =
+    await getTaskDefinitionAndTaskArguments(allUnparsedCLAs);
+
+  switch (taskDefinition.name) {
+    case "set":
+      return set(taskArguments.var, taskArguments.value);
+    case "get":
+      return get(taskArguments.var);
+    case "list":
+      return list();
+    case "delete":
+      return del(taskArguments.var);
+    case "path":
+      return path();
+    case "setup":
+      return setup(configPath);
+    default:
+      throw new HardhatError(ERRORS.VARS.INVALID_TASK_DEFINITION_NAME, {
```

Comment:
Sorry if my previous comment was too ambiguous. I just meant adding a `console.error` with an explanation. I don't think it's necessary to add a new error type for this.